odoo/fields.py

```diff
@@ -1536,11 +1543,17 @@ class _String(Field):
     translate = False                   # whether the field is translated
     prefetch = None
     unaccent = True
+    pattern = None                      # regex to validate an input
+    pattern_flags = None                # flags used for the regex validation
 
     def __init__(self, string=Default, **kwargs):
         # translate is either True, False, or a callable
         if 'translate' in kwargs and not callable(kwargs['translate']):
             kwargs['translate'] = bool(kwargs['translate'])
+        if 'pattern' in kwargs and 'help' not in kwargs:
+            _logger.warning('field %s,%s has a pattern defined without any `help`', self.model_name, self.name)
+        if 'pattern' in kwargs and 'pattern_flags' not in kwargs:
+            _logger.warning('field %s,%s has a pattern defined without `pattern_flags`', self.model_name, self.name)
```

Comment:
When looking at the examples, my suggestion looks very cumbersome.
So now I think you were right, `pattern_flags` should simply default to `0`...
Sorry for the extra work.